From f5c38ad858ccd8124ba611c74792620a9c70857e Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Fri, 20 Jul 2018 16:18:52 +0800
Subject: [PATCH 4466/5242] MLK-18992-2 drm/imx: hdp: Don't set pixel link MST
 address

commit  95b7c31b2335a0413955e90eaa6200e67ebcc3f7 from
https://source.codeaurora.org/external/imx/linux-imx.git

The pixel link MST address is set by the display controller driver,
so let's remove the redundant setting from the hdp driver.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
(cherry picked from commit f0f0a1970f2b34ad52fa9cfcb1a4de503fb231d6)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/hdp/imx-hdp.c |    4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/gpu/drm/imx/hdp/imx-hdp.c b/drivers/gpu/drm/imx/hdp/imx-hdp.c
index 4a47f83..4725d02 100644
--- a/drivers/gpu/drm/imx/hdp/imx-hdp.c
+++ b/drivers/gpu/drm/imx/hdp/imx-hdp.c
@@ -125,8 +125,6 @@ int imx8qm_pixel_link_init(state_struct *state)
 		return -EINVAL;
 	}
 
-	/* config dpu1 di0 to hdmi/dp mode */
-	sc_misc_set_control(hdp->ipcHndl, SC_R_DC_0, SC_C_PXL_LINK_MST1_ADDR, 1);
 	sc_misc_set_control(hdp->ipcHndl, SC_R_DC_0, SC_C_PXL_LINK_MST1_VLD, 1);
 	sc_misc_set_control(hdp->ipcHndl, SC_R_DC_0, SC_C_SYNC_CTRL0, 1);
 
@@ -137,8 +135,6 @@ void imx8qm_pixel_link_deinit(state_struct *state)
 {
 	struct imx_hdp *hdp = state_to_imx_hdp(state);
 
-	/* config dpu1 di0 to default mode */
-	sc_misc_set_control(hdp->ipcHndl, SC_R_DC_0, SC_C_PXL_LINK_MST1_ADDR, 0);
 	sc_misc_set_control(hdp->ipcHndl, SC_R_DC_0, SC_C_PXL_LINK_MST1_VLD, 0);
 	sc_misc_set_control(hdp->ipcHndl, SC_R_DC_0, SC_C_SYNC_CTRL0, 0);
 
-- 
1.7.9.5

